src/pages/login/SetupSsoLogin.tsx

```diff
@@ -0,0 +1,44 @@
+import type { FC } from "react";
+import { Link, Navigate } from "react-router-dom";
+import { useAuth } from "context/auth";
+import { Icon, Spinner, CustomLayout } from "@canonical/react-components";
+import SsoProviderSelector from "components/SsoProviderSelector/SsoProviderSelector";
+import RenameHeader from "components/RenameHeader";
+
+const SetupSsoLogin: FC = () => {
+  const { isAuthenticated, isAuthLoading } = useAuth();
+
+  if (isAuthLoading) {
+    return <Spinner className="u-loader" text="Loading..." isMainComponent />;
+  }
+
+  if (isAuthenticated) {
+    return <Navigate to="/ui" replace={true} />;
+  }
+
+  return (
+    <CustomLayout
+      contentClassName="login"
+      header={
+        <RenameHeader
```

Comment:
This seems like a wrong usage of the component, as there is no renaming here. Maybe we can use the same modal interaction as on the identity page?